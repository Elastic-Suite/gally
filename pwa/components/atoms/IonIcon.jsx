import React from 'react'

/* Creation of special props to be clean on Typescripts files
 * See: https://github.com/ionic-team/ionicons
 */
const IonIcon = (props) => {
    return <ion-icon {...props} />
}

export default IonIcon
