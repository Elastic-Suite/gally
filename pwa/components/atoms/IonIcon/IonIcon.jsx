import React from 'react'
import home2 from '~/assets/images/home2.svg'

export const customIcons = [
    'dashboard',
    'analyze',
    'merchandize',
    'monitoring',
    'settings',
]

/*
 * Creation of special props to be clean on Typescripts files
 * See: https://github.com/ionic-team/ionicons
 *
 * Setup switch for special names that need svg or have another name in ion-icons
 */
const IonIcon = (props) => {
    const propsCleaned = { ...props }
    delete propsCleaned.name
    switch (props.name) {
        case 'dashboard':
            return <ion-icon src={home2.src} {...propsCleaned} />
        case 'analyze':
            return <ion-icon name={'analytics'} {...propsCleaned} />
        case 'merchandize':
            return <ion-icon name={'funnel-outline'} {...propsCleaned} />
        case 'monitoring':
            return <ion-icon name={'list'} {...propsCleaned} />
        case 'settings':
            return <ion-icon name={'settings-outline'} {...propsCleaned} />
        default:
            return <ion-icon name={props.name} {...propsCleaned} />
    }
}

export default IonIcon
