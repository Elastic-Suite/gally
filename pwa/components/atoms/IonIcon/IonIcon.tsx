/* eslint-disable @typescript-eslint/no-namespace */
import { CSSProperties, MouseEvent } from 'react'
import { JSX as IonIconJSX } from 'ionicons'
import { JSXBase } from 'ionicons/dist/types/stencil-public-runtime'

export type IIonIconProps = Omit<
  IonIconJSX.IonIcon & JSXBase.HTMLAttributes<HTMLIonIconElement>,
  'style' | 'onClick'
> & {
  style?: CSSProperties
  onClick?: (event: MouseEvent<IIonIconProps>) => void
}

declare global {
  namespace JSX {
    // eslint-disable-next-line @typescript-eslint/naming-convention
    interface IntrinsicElements {
      'ion-icon': IIonIconProps
    }
  }
}

interface IProps extends IIonIconProps {
  name: string
  tooltip?: boolean
}

export const customIcons = [
  'dashboard',
  'analyze',
  'merchandize',
  'monitoring',
  'settings',
]

/*
 * Creation of special props to be clean on Typescripts files
 * See: https://github.com/ionic-team/ionicons
 *
 * Setup switch for special names that need svg or have another name in ion-icons
 */
function IonIcon(props: IProps): JSX.Element {
  const { name, tooltip, ...iconProps } = props
  switch (name) {
    case 'dashboard':
      return <ion-icon src="/images/home2.svg" {...iconProps} />
    case 'analyze':
      return <ion-icon name="analytics" {...iconProps} />
    case 'merchandize':
      return <ion-icon name="funnel-outline" {...iconProps} />
    case 'monitoring':
      return <ion-icon name="list" {...iconProps} />
    case 'settings':
      return <ion-icon name="settings-outline" {...iconProps} />
    case 'information':
      return (
        <ion-icon
          name="information-circle-outline"
          style={
            tooltip
              ? {
                  color: '#8187B9',
                  width: '20px',
                  height: '20px',
                  display: 'flex',
                }
              : { ...iconProps }
          }
        />
      )
    default:
      return <ion-icon name={name} {...iconProps} />
  }
}

export default IonIcon
