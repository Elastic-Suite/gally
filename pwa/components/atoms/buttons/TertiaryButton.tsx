import { ReactNode } from 'react'
import { ButtonProps } from '@mui/material/Button'
import { styled } from '@mui/system'

import CommonStyledButton from '~/components/atoms/buttons/CommonButtonStyle'

const TertiaryStyledButton = styled(CommonStyledButton)(({ theme }) => ({
  color: theme.palette.colors.neutral['900'],
  background: 'none',
  boxShadow: 'none',
  '&::before': {
    background: theme.palette.colors.neutral['200'],
    boxShadow: 'none',
  },
  '&:hover': {
    background: 'none',
    boxShadow: 'none',
  },
  '&& .MuiTouchRipple-child': {
    backgroundColor: theme.palette.colors.neutral['300'],
  },
}))

interface IProps extends ButtonProps {
  children: ReactNode
}

function TertiaryButton(props: IProps): JSX.Element {
  const { children, ...buttonProps } = props

  return (
    <TertiaryStyledButton {...buttonProps} variant="contained">
      {children}
    </TertiaryStyledButton>
  )
}

export default TertiaryButton
