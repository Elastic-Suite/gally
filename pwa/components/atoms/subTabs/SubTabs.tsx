import { ReactChild, useState } from 'react'
import { keyframes, styled } from '@mui/system'

const CustomRoot = styled('div')(({ theme }) => ({
  display: 'flex',
  gap: theme.spacing(4),
}))

const CustumRootSubTabs = styled('div')({
  border: '1px solid #E2E6F3',
  borderRadius: 8,
  display: 'flex',
  flexDirection: 'column',
  minWidth: '232px',
  boxSizing: 'border-box',
  height: 'max-content',
})

const CustomSubTabs = styled('div')(({ theme }) => ({
  padding: theme.spacing(1),
  borderRadius: 8,
  marginLeft: theme.spacing(2),
  marginRight: theme.spacing(2),
  fontSize: '14px',
  lineHeight: '20px',
  fontWeight: '500',
  color: theme.palette.colors.neutral[600],
  transition: 'all 500ms',
  cursor: 'pointer',

  '&:nth-child(1)': {
    marginTop: theme.spacing(2),
  },

  '&:nth-last-child(1)': {
    marginBottom: theme.spacing(2),
  },
  '&:hover': {
    backgroundColor: theme.palette.colors.secondary[100],
  },
}))

const CustomSubTabsActive = styled(CustomSubTabs)(({ theme }) => ({
  color: theme.palette.colors.secondary[600],
  position: 'relative',
  '&::after': {
    content: "''",
    position: 'absolute',
    right: '-16px',
    top: '50%',
    transform: 'translateY(-50%)',
    height: '32px',
    width: 3,
    opacity: 0,
    background: theme.palette.menu.active,
    boxShadow: '-2px 0px 4px rgba(63, 50, 230, 0.2)',
    borderRadius: '5px 0px 0px 5px',
    animation: `${opacity} 1500ms forwards`,
  },
}))

const opacity = keyframes`
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
`

interface IProps {
  labels: any
  contents: ReactChild[]
}

function SubTabs({ labels, contents }: IProps): JSX.Element {
  const [active, setActive] = useState(0)

  return (
    <CustomRoot>
      <CustumRootSubTabs>
        {labels.map((item: string, key: number) =>
          key === active ? (
            <CustomSubTabsActive
              key={item}
              onClick={(): void => setActive(key)}
            >
              {item}
            </CustomSubTabsActive>
          ) : (
            <CustomSubTabs key={item} onClick={(): void => setActive(key)}>
              {item}
            </CustomSubTabs>
          )
        )}
      </CustumRootSubTabs>
      {contents.map(
        (item: string, key: number) =>
          key === active && (
            <div style={{ width: '100%' }} key={item}>
              {item}
            </div>
          )
      )}
    </CustomRoot>
  )
}

export default SubTabs
