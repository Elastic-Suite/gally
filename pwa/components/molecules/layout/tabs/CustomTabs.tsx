import { SyntheticEvent, useState } from 'react'
import Tabs from '@mui/material/Tabs'
import Tab from '@mui/material/Tab'
import Box from '@mui/material/Box'
import TabPanel from './TabPanel'
import { a11yProps } from './a11yProps'

interface IProps {
  labels: Array<string>
  contents: string[]
}
export default function CustomTabs({ labels, contents }: IProps): JSX.Element {
  const [activeTabIndex, setActiveTabIndex] = useState(0)
  const handleChange = (event: SyntheticEvent, newValue: number): void => {
    event.preventDefault()
    setActiveTabIndex(newValue)
  }

  return (
    <Box sx={{ width: '100%', marginTop: '-12px' }}>
      <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
        <Tabs
          value={activeTabIndex}
          onChange={handleChange}
          aria-label="basic tabs example"
        >
          {labels.map((item: string, key: number) => (
            <Tab
              key={item}
              label={item}
              {...a11yProps('simple-tabpanel', key)}
            />
          ))}
        </Tabs>
      </Box>
      {contents.map((item: string, key: number) => (
        <TabPanel key={item} value={activeTabIndex} index={key}>
          {item}
        </TabPanel>
      ))}
    </Box>
  )
}
