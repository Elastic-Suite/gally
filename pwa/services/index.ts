export * from './api'
export * from './authProvider'
export * from './format'
export * from './hydra'
export * from './jsonld'
export * from './local'
export * from './table'
export * from './tests'
