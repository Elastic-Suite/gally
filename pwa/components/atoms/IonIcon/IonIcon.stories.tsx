import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { icons } from '~/constants';

import IonIcon from './IonIcon';

export default {
  title: 'Atoms/IonIcon',
  component: IonIcon,
  argTypes: {
    name: {
      options: icons,
      control: { type: 'select' },
    },
  }
} as ComponentMeta<typeof IonIcon>;

const Template: ComponentStory<typeof IonIcon> = (args) => <IonIcon {...args} />;

export const Default = Template.bind({});
Default.args = {
  name: 'analytics'
};
