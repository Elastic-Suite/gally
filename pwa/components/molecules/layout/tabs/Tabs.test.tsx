import CustomTabs from '~/components/molecules/layout/tabs/CustomTabs'
import SubTabs from '~/components/atoms/subTabs/SubTabs'
import TertiaryButton from '~/components/atoms/buttons/TertiaryButton'
import IonIcon from '~/components/atoms/IonIcon/IonIcon'
import { renderWithProviders } from '~/services'

describe('TabsAndSubTabs', () => {
  it('match snapshot', () => {
    const { container } = renderWithProviders(
      <CustomTabs
        labels={['Scope', 'Searchable and filtrable attributes']}
        contents={[
          <SubTabs
            key="labels"
            labels={[
              'Catalogs',
              'Actives locales',
              'Actives locales 1',
              'Actives locales 2',
            ]}
            contents={[
              <TertiaryButton
                key="contents"
                size="medium"
                endIcon={
                  <IonIcon name="add-outline" style={{ fontSize: 24 }} />
                }
              >
                Button text
              </TertiaryButton>,
              'Hello World Deux',
              'Hello World Trois',
              'Hello World Quatre',
            ]}
          />,
          'Content of Searchable and filtrable attributes',
        ]}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
