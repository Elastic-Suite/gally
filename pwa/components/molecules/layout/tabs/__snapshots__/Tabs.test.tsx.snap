// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsAndSubTabs match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-e3slrw"
  >
    <div
      class="MuiBox-root css-1gsv261"
    >
      <div
        class="MuiTabs-root css-1xt4903-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-kqhgxw-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            aria-label="basic tabs example"
            class="MuiTabs-flexContainer css-1pyy021-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-controls="simple-tabpanel-0"
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-v7gylc-MuiButtonBase-root-MuiTab-root"
              id="simple-tab-0"
              role="tab"
              tabindex="0"
              type="button"
            >
              Scope
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-controls="simple-tabpanel-1"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-v7gylc-MuiButtonBase-root-MuiTab-root"
              id="simple-tab-1"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Searchable and filtrable attributes
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1bftaej-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </div>
    <div
      aria-labelledby="simple-tab-0"
      id="simple-tabpanel-0"
      role="tabpanel"
    >
      <div
        class="MuiBox-root css-ed63to"
      >
        <div
          class="css-13t3iqc"
        >
          <div
            class="css-1n8gd2i"
          >
            <div
              class="css-9wk4uw"
            >
              Catalogs
            </div>
            <div
              class="css-56xlw1"
            >
              Actives locales
            </div>
            <div
              class="css-56xlw1"
            >
              Actives locales 1
            </div>
            <div
              class="css-56xlw1"
            >
              Actives locales 2
            </div>
          </div>
          <div>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-tvac2b-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Button text
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium css-1gnd1fd-MuiButton-endIcon"
              >
                <ion-icon
                  name="add-outline"
                  style="font-size: 24px;"
                />
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="simple-tab-1"
      hidden=""
      id="simple-tabpanel-1"
      role="tabpanel"
    />
  </div>
</div>
`;
